### DISCUSSION TITLE
When we use this controller, should we assign it to a public variable?

### DISCUSSION THREAD

**Discussion by heyMP** [RedHat-UX/red-hat-design-system#752] [General]
*The following was a discussion taken from internal GChat thread.  Moving here for visibility.*

**Michael Potter:**

@Benny Powers Looking at the internal controllers PR.  When we use this controller, should we assign it to a public variable?  So that implementors could use it when building forms. 

`#internals = new InternalsController(this);`

to this:

`internals = new InternalsController(this);`

![unnamed (9)](https://user-images.githubusercontent.com/3428964/222486603-d5c47f01-0fd2-4e1b-a037-d9547aa30fe8.png)

ref: https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reportValidity

**Benny Powers**

internals should be internal i think

if devs using the component are having trouble using it when buildling a form, that's on us as component devs

should be `button.reportValidity()` instead. that could eventually be a mixin or a base class or something

**Comment by heyMP** [comment]
This is a unique instance where we want the best of both worlds.  We want to compartmentalize all of the magic of setting up the element internals but put it into a discrete Lit controller, however, we also want it to create public api methods, getters, and setters on the class implementing the controller so that it more accurately reflects the existing api spec.

Sounds like a job for either mixin or base class as Benny suggested.

**Reply by brianferry** [reply]
Would we want to slowly introduce public getters / methods based on element internals for different use cases or are we thinking that we'd have all the properties be accessible through the mixin / base class just not the actual internals object itself?
